govcd/metadata_test.go

```diff
@@ -181,3 +182,165 @@ func (vcd *TestVCD) Test_DeleteMetadataOnVm(check *C) {
 		}
 	}
 }
+
+func (vcd *TestVCD) Test_DeleteMetadataOnVAppTemplate(check *C) {
+
+	fmt.Printf("Running: %s\n", check.TestName())
+	cat, err := vcd.org.FindCatalog(vcd.config.VCD.Catalog.Name)
+	if err != nil {
+		check.Skip("Test_DeleteMetadataOnCatalogItem: Catalog not found. Test can't proceed")
+	}
+
+	if vcd.config.VCD.Catalog.CatalogItem == "" {
+		check.Skip("Test_DeleteMetadataOnCatalogItem: Catalog Item not given. Test can't proceed")
+	}
+
+	catItem, err := cat.FindCatalogItem(vcd.config.VCD.Catalog.CatalogItem)
+	check.Assert(err, IsNil)
+
+	vAppTemplate, err := catItem.GetVAppTemplate()
+	check.Assert(err, IsNil)
+
+	// Add metadata
+	_, err = vAppTemplate.AddMetadata("key2", "value2")
+	check.Assert(err, IsNil)
+
+	// Remove metadata
+	err = vAppTemplate.DeleteMetadata("key2")
+	check.Assert(err, IsNil)
+
+	metadata, err := vAppTemplate.GetMetadata()
+	check.Assert(err, IsNil)
+	for _, k := range metadata.MetadataEntry {
+		if k.Key == "key2" {
+			check.Errorf("metadata.MetadataEntry should not contain key: 'key2'")
+		}
+	}
+}
+
+func (vcd *TestVCD) Test_AddMetadataOnVAppTemplate(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+	cat, err := vcd.org.FindCatalog(vcd.config.VCD.Catalog.Name)
+	if err != nil {
+		check.Skip("Test_DeleteMetadataOnCatalogItem: Catalog not found. Test can't proceed")
+	}
+
+	if vcd.config.VCD.Catalog.CatalogItem == "" {
+		check.Skip("Test_DeleteMetadataOnCatalogItem: Catalog Item not given. Test can't proceed")
+	}
+
+	catItem, err := cat.FindCatalogItem(vcd.config.VCD.Catalog.CatalogItem)
+	check.Assert(err, IsNil)
+
+	vAppTemplate, err := catItem.GetVAppTemplate()
+	check.Assert(err, IsNil)
+
+	// Check how much metaData exist
+	metadata, err := vAppTemplate.GetMetadata()
+	check.Assert(err, IsNil)
+	existingMetaDataCount := len(metadata.MetadataEntry)
+
+	// Add metadata
+	_, err = vAppTemplate.AddMetadata("key", "value")
+	check.Assert(err, IsNil)
+
+	// Check if metadata was added correctly
+	metadata, err = vAppTemplate.GetMetadata()
+	check.Assert(err, IsNil)
+	check.Assert(len(metadata.MetadataEntry), Equals, existingMetaDataCount+1)
+	var foundEntry *types.MetadataEntry
+	for _, entry := range metadata.MetadataEntry {
+		if entry.Key == "key" {
+			foundEntry = entry
+		}
+	}
+	check.Assert(foundEntry, NotNil)
+	check.Assert(foundEntry.Key, Equals, "key")
+	check.Assert(foundEntry.TypedValue.Value, Equals, "value")
+
+	err = vAppTemplate.DeleteMetadata("key")
+	check.Assert(err, IsNil)
+}
+
+func (vcd *TestVCD) Test_DeleteMetadataOnMediaItem(check *C) {
+	//prepare media item
+	skipWhenMediaPathMissing(vcd, check)
+	itemName := "TestDeleteMediaMetaData"
+
+	org, err := GetAdminOrgByName(vcd.client, vcd.org.Org.Name)
+	check.Assert(org, Not(Equals), AdminOrg{})
```

Comment:
Calling an about-to-be-deprecated-function. This will need a clean-up after PR #217 is merged

Reply:
fixed